File: linmour-common/src/main/java/com/linmour/common/dtos/Result.java
package com.linmour.common.dtos;

import lombok.Data;
import lombok.ToString;

/**
 * @Classname Result
 * @Description 通用返回类型
 * @Date 2023/7/17 11:01
 * @Created by linmour
 */


@Data
@ToString
public class Result<T> {

    /**
     * 响应编码,0为正常,-1错误
     */
    private int code;

    /**
     * 响应提示信息
     */
    private String msg;

    /**
     * 响应内容
     */
    private T result;


    public Result() {
        this(0, "success");
    }

    public Result(int code, String msg) {
        this.code = code;
        this.msg = msg;
    }

    /**
     * 错误信息的封装
     *
     * @param msg
     * @param <T>
     * @return
     */
    public static <T> com.linmour.common.dtos.Result<T> validfail(String msg) {
        com.linmour.common.dtos.Result<T> response = new com.linmour.common.dtos.Result<T>();
        response.setCode(-1);
        response.setMsg(msg);
        return response;
    }
    public static <T> com.linmour.common.dtos.Result<T> validfail(T result, String msg) {
        com.linmour.common.dtos.Result<T> response = new com.linmour.common.dtos.Result<T>();
        response.setCode(-1);
        response.setResult(result);
        response.setMsg(msg);
        return response;
    }



    /**
     * 添加正常响应数据（包含响应内容）
     *
     * @return RestResponse Rest服务封装相应数据
     */
    public static <T> com.linmour.common.dtos.Result<T> success(T result) {
        com.linmour.common.dtos.Result<T> response = new com.linmour.common.dtos.Result<T>();
        response.setResult(result);
        return response;
    }
    public static <T> com.linmour.common.dtos.Result<T> success(T result, String msg) {
        com.linmour.common.dtos.Result<T> response = new com.linmour.common.dtos.Result<T>();
        response.setResult(result);
        response.setMsg(msg);
        return response;
    }

    /**
     * 添加正常响应数据（不包含响应内容）
     *
     * @return RestResponse Rest服务封装相应数据
     */
    public static <T> com.linmour.common.dtos.Result<T> success() {
        return new com.linmour.common.dtos.Result<T>();
    }


    public Boolean isSuccessful() {
        return this.code == 0;
    }

}

File: linmour-order-module-go/server/models/Result.go
package models

// Result 用于定义响应结构体
type Result struct {
	Code int         `json:"code"` // 响应编码
	Msg  string      `json:"msg"`  // 响应提示信息
	Data interface{} `json:"data"` // 响应内容
}

// NewResult 创建一个新的Result实例
func NewResult(code int, msg string, data interface{}) *Result {
	return &Result{
		Code: code,
		Msg:  msg,
		Data: data,
	}
}

// Error 创建一个错误响应
func Error(msg string) *Result {
	return NewResult(201, msg, nil)
}

// ErrorWithData 创建一个带有数据的错误响应
func ErrorWithData(data interface{}, msg string) *Result {
	return NewResult(201, msg, data)
}

// Success 创建一个成功的响应
func Success(data interface{}) *Result {
	return NewResult(200, "success", data)
}
